sympy/polys/matrices/domainmatrix.py

```diff
@@ -1690,3 +1692,74 @@ def unify_eq(A, B):
         if A.domain != B.domain:
             A, B = A.unify(B)
         return A == B
+
+    def lll(A, delta: Rational = Rational(3, 4)) -> 'DomainMatrix':
```

Comment:
Rational should not be used here either. The float 0.75 is exact and the algorithm output is somewhat heuristic anyway. I think that we can allow float or QQ here.

Separately I wonder if it would be good to have a way to have the whole algorithm use float instead of QQ. I think that's what one of the flint methods does and I suspect that's why it's asymptotically faster than the method here.

Reply:
Unless there is an objection to this, I think this should go into the float ticket, it seems in scope for that. Right now the algorithm runs QQ throughout.

Reply:
The problem is about levels and circularity within the codebase. Here polys should really be considered to be at a level below the core symbolic classes and Basic. DomainMatrix, DDM and the domains should all be defined and implemented without depending on any part of Basic. In principle it should be possible to make the polys module a separate library that the rest of SymPy just depends on (actually it would be better if it was a separate library). The change here makes this module import from core which I consider to be circular even if it does not immediately cause a circular import error. We can use `QQ` here but not `Rational`.

Reply:
Ok, I will change it to QQ